src/bots/inmessage.py

```diff
@@ -1714,6 +1723,128 @@ def _getrootid(self):
+                newnode = self._doavroobject({value[0]: value[1]}, key)
+                if newnode:
+                    newnode.record['AVRO_RECORD_NAME'] = value[0]
+                    thisnode.append(newnode)
+            elif isinstance(value, UUID):  # uuid will be serialized by fastavro as a UUID
+                thisnode.record[key] = unicode(value)
+            else:
+                if self.ta_info['checkunknownentities']:
+                    raise botslib.InMessageError(_('[J55]: Key "%(key)s" value "%(value)s": is not string, list or dict.'),
+                                                 {'key': key, 'value': value})
+                thisnode.record[key] = unicode(value)
+        if len(thisnode.record) == 2 and not thisnode.children:
+            return None  # node is empty...
+        return thisnode
+
+    def _getrootid(self):
+        return self.defmessage.structure[0][ID]
+
+    def _isavromap(self, name):
+        exists = False
+        for item in self.defmessage.recorddefs[name]:
+            if(item[ID] == 'AVRO_MAP'):
+                exists = True
+        return exists
+
+    def _doavrolist(self, avroobject, name):
+        lijst = []  # initialise empty list, used to append a listof (converted) avro objects
+        for i in avroobject:
+            if isinstance(i, dict):  # check list item is dict/object
+                newnode = self._doavroobject(i, name)
+                if newnode:
+                    lijst.append(newnode)
+            elif isinstance(i, list):  # check list item is list:
+                newnode = self._doavrolist(i, name)
+                if newnode:
+                    lijst.append(newnode)
+            else:
+                newnode = self._doavroobject({'input': i, 'AVRO_ARRAY': 'True'}, name)
+                if newnode:
+                    lijst.append(newnode)
+        return lijst
+
+    def _doavromap(self, avroobject, name):
+        lijst = []
+        for key, value in avroobject.items():
+            if value is None:
+                continue
+            elif isinstance(value, basestring):  # avro field; map to field in node.record
+                ## for generating grammars: empty strings should generate a field
+                if value:  # use only if string has a value.
+                    newnode = self._doavroobject({'key': key, 'value': value, 'AVRO_MAP': 'True'}, name)
+                    if newnode:
+                        lijst.append(newnode)
+            elif isinstance(value, (int, long, float)):  # avro field; map to field in node.record
+                newnode = self._doavroobject({'key': key, 'value': value, 'AVRO_MAP': 'True'}, name)
+                if newnode:
+                    lijst.append(newnode)
+            else:
+                raise botslib.InMessageError(_('[J54]: Avromap values only supports primitive types'))
+        return lijst
+
+    def _readcontent_edifile(self):
```

Comment:
If this is not an interface method, we should rename it I think to _readcontent_avrofile